src/Blaster.WebApi/Features/Shared/HttpReponseHelper.cs

```diff
@@ -12,39 +12,25 @@ public static void EnsureSuccessStatusCode(HttpResponseMessage response)
 		{
 			if (response.StatusCode == HttpStatusCode.Unauthorized)
 			{
-				throw new UnauthroizedException();
+				throw new UnauthorizedException();
 			}
 
 			response.EnsureSuccessStatusCode();
 		}
 
 		public static async Task MapStatusCodeToException(HttpResponseMessage response)
 		{
-			if (response.StatusCode == HttpStatusCode.Unauthorized)
-			{
-				throw new UnauthroizedException();
-			}
-
-			if (HttpStatusCode.Conflict == response.StatusCode)
+			switch (response.StatusCode)
 			{
-				var payload = await response.Content.ReadAsStringAsync();
-				throw new RecoverableUpstreamException(response.StatusCode, payload);
+				case HttpStatusCode.Unauthorized:
+					throw new UnauthorizedException();
+				case HttpStatusCode.Conflict:
+				case HttpStatusCode.UnprocessableEntity:
+				{
+					var payload = await response.Content.ReadAsStringAsync();
+					throw new RecoverableUpstreamException(response.StatusCode, payload);
+				}
```

Comment:
Missing default "unknown error"-like-ish thingy!

Reply:
I looked it over and the switch does not need a default behavior because the switch operates on HttpStatusCode which can't be in a invalid state. all HttpStatusCodes not covered in the switch should allow the control flow to continue normally instead of throwing a exception as the described cases of the switch.

Reply:
Cool! Good catch